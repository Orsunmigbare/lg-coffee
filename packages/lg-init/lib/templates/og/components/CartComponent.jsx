/* eslint-disable no-unused-expressions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import Router from 'next/router';
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { calc } from 'lg-calc';
import Modal from 'react-bootstrap/Modal';
import CheckoutForm from './CheckoutForm';
import CheckoutPreview from './CheckoutPreview';

const CartComponent = ({ cart }) => {
  const [isCartEmpty, setCartToEmpty] = useState();
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  useEffect(() => {
    cart ? setCartToEmpty(false) : setCartToEmpty(true);
  }, [cart]);

  return (
    <>
      <div variant="primary" onClick={handleShow}>
        <img
          src="/cart.png"
          alt="Checkout"
          style={{
            height: '20px',
            cursor: 'pointer',
          }}
        />
      </div>

      <Modal show={show} onHide={handleClose}>
        {isCartEmpty ? (
          <h2 className="m-4">Nothing in cart</h2>
        ) : (
          <div className="d-flex flex-column align-items-center">
            <div>
              <div className="d-flex flex-column">
                <button
                  type="button"
                  onClick={handleClose}
                  className="align-self-end mt-4 mr-4 btn btn-outline-dark"
                  style={{
                    border: 'none',
                    background: 'none',
                    fontSize: '16px',
                    fontFamily: 'Montserrat',
                  }}
                >
                  Go Back
                </button>
              </div>
              <CheckoutPreview cart={cart} />
              <CheckoutForm
                cart={cart}
                price={calc(cart)}
                onSuccessfulCheckout={() =>
                  Router.push('/checkout/order-confirmation')
                }
              />
            </div>
          </div>
        )}
      </Modal>
    </>
  );
};

CartComponent.propTypes = {
  // eslint-disable-next-line react/require-default-props
  cart: PropTypes.shape({}),
  data: PropTypes.shape({}).isRequired,
};

export default CartComponent;
