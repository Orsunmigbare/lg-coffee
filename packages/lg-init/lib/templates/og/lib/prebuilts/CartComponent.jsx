import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/Modal';
import CartPlaceholder from '../components/CartPlaceholder';
import CartModal from './CartModal';

const CartComponent = ({ cart, handleUpdatedCartInState }) => {
  const [isCartEmpty, setCartToEmpty] = useState(true);

  // modal funcs
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  useEffect(() => {
    // eslint-disable-next-line no-unused-expressions
    cart ? setCartToEmpty(false) : setCartToEmpty(true);
  }, [cart]);

  const handleEmptyCartNotice = () => setCartToEmpty(true);

  return (
    <>
      <CartPlaceholder handleShow={handleShow} isCartEmpty={isCartEmpty} />

      <Modal show={show} onHide={handleClose}>
        {isCartEmpty ? (
          <h2 className="m-4">Nothing in cart</h2>
        ) : (
          <CartModal
            cart={cart}
            handleClose={handleClose}
            handleEmptyCartNotice={handleEmptyCartNotice}
            handleUpdatedCartInState={(updatedCart) =>
              handleUpdatedCartInState(updatedCart)
            }
          />
        )}
      </Modal>
    </>
  );
};

CartComponent.propTypes = {
  cart: PropTypes.shape({}),
  data: PropTypes.shape({}).isRequired,
  handleUpdatedCartInState: PropTypes.func.isRequired,
};

CartComponent.defaultProps = {
  cart: {},
};

export default CartComponent;
