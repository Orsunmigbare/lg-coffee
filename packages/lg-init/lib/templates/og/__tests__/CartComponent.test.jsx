import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CartComponent from '../components/CartComponent';

const product = {
  name: 'product',
  description: 'this is an awesome product you should buy 2',
  price: '$2.87',
  sku: 'abc_123',
  images: [
    '/path_01.png',
    '/path_02.png',
    '/path_03.png',
  ],
  qty: 12,
};

const handleUpdatedCartInState = () => {};

it('CheckoutError component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <CartComponent 
      cart={{product}} 
      handleUpdatedCartInState={handleUpdatedCartInState} 
    />
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('CheckoutError component renders unchanged', () => {
  const tree = renderer.create(
    <CartComponent 
      cart={{product}} 
      handleUpdatedCartInState={handleUpdatedCartInState} 
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
