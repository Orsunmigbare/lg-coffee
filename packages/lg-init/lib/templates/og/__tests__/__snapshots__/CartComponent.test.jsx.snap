// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutError component renders unchanged 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "none",
      "border": "0",
    }
  }
  type="button"
>
  <img
    alt="Checkout"
    src="/cart.png"
    style={
      Object {
        "cursor": "pointer",
        "height": "20px",
      }
    }
  />
</button>
`;
